docs/guides/es2k/deploying-p4-programs.md

```diff
@@ -0,0 +1,139 @@
+# Deploying P4 Programs for ES2K
+
+## 1. Overview
+
+This document explains how to deploy the P4 artifacts on hardware flexible
+pipeline (FXP) and software control plane `infrap4d`.
+
+Note: This document assumes that you have followed
+[Compiling P4 programs guide](https://github.com/nupuruttarwar/networking-recipe/blob/main/docs/guides/es2k/compiling-p4-programs.md)
+to compile and generate artifacts required for deployment.
+
+## 2. Deploying on Hardware Flexible Pipeline (FXP)
+
+Data Path Control Plane (DPCP) starts with a default P4 package. To load a
+custom P4 package follow below steps:
+
+### 2.1 Interrupt default startup routine
+
+Reboot IMC and type ``N`` when the following message is shown on IMC console::
+
+```text
+start ipumgmtd and auxiliary script [Y/N] \
+(default start both automatically after 10 seconds)?
+```
+
+### 2.2 Copy the custom P4 package
+
+Copy the custom P4 package (.pkg) in `/etc/dpcp/package` directory and
+overwrite the `default_pkg.pkg`.
+
+For example, if replacing `default-pkg.pkg` with `simple_l2_demo.pkg`
+
+```bash
+root@mev-imc:/etc/dpcp/package# ls -lrt /etc/dpcp/package/
+total 2364
+-rw-r--r-- 1 root root  963032 Jan  1 04:56 simple_l2_demo.pkg
+-rw-r--r-- 1 root root 1450456 Jun  8  2023 e2100-default-1.0.3.0.pkg
+drwxr-xr-x 2 root root       0 Jun  8  2023 runtime_files
+drwxr-xr-x 3 root root       0 Jun  8  2023 ref_pkg
+lrwxrwxrwx 1 root root      25 Jun  8  2023 default_pkg.pkg -> e2100-default-1.0.3.0.pkg
+root@mev-imc:/etc/dpcp/package# cp simple_l2_demo.pkg default_pkg.pkg
+```
+
+### 2.3 Start the IMC
+
+Run the IMC start-up script.
+
+```bash
+root@mev-imc:~# /etc/init.d/run_default_init_app
+```
+
+## 3. Deploying on Software Control Plane (infrap4d)
+
+On ACC, use `tdi_pipeline_builder` available under `/opt/p4/p4-cp-nws/bin` to
+generate forwarding pipeline binary.
+
+Note: We will be using artifacts generated by following
+[Compilling P4 programs guide](https://github.com/nupuruttarwar/networking-recipe/blob/main/docs/guides/es2k/compiling-p4-programs.md).
+
+### 3.1 Prepare configuration files
+
+Create tofino.bin
+
+```bash
+export OUTPUT_DIR=/usr/share/mev_reference_p4_files/simple_l3_l4_pna
+cd $OUTPUT_DIR
+touch tofino.bin
+```
+
+Handcraft the configuration file `/usr/share/stratum/es2k_skip_p4.conf` with
```

Comment:
isnt the directory path 
"/usr/share/stratum/**es2k**/es2k_skip_p4.conf" ?

is it changed by any chance?